packages/flame/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,179 @@
+import 'package:flutter/gestures.dart';
+
+import '../../extensions.dart';
+
+abstract class BaseInfo<T> {
+  final T raw;
+
+  BaseInfo(this.raw);
+}
+
+abstract class PositionInfo<T> extends BaseInfo<T> {
+  final Vector2 position;
+  final Vector2 gameWidgetPosition;
+  final Vector2 globalPosition;
```

Comment:
a thought: should we have a Coordinate class that wrap these three? then

```
PositionInfo { coordinate }
TapInfo extends PositionInfo {}
SomeWeirdInfo extends BaseInfo { coordinate1, coordinate2 }
```

not sure if worth the indirection. but that would force users to always consider witch coordinate system to get the vector2 from

Reply:
I don't see that much value on doing that as I think it would not change much neither for us or the final dev.

But I also don't mind implementing it if you folks prefer that.

Reply:
it was more hypothetical because I don't know how many (if any) events are not PositionInfo and have more than one translatable coordinates. for example if there was an event with 3 positions having 9 fields would be cumbersome